src/js/modules/articles/actions.js

```diff
@@ -1,11 +1,55 @@
 import axios from 'axios';
 import * as t from './actionTypes';
+import STUY_SPEC_API from '../../constants';
 
-const apiUrl = 'http://api.icndb.com/jokes/random';
-
+//TODO: need a way to call the fetch function
 export const fetch = () => {
-	return {
-		type: t.FETCH_ARTICLE,
-		payload: axios.get(apiUrl),
-	};
+  return (dispatch, getState) => {
+    dispatch({ type: t.FETCH_ARTICLE_PENDING });
+    //axios.get(apiUrl, {'headers': {'X-Key-Inflection': 'camel'}}) TODO: wait until Nic uses olive branch
+    axios.get(STUY_SPEC_API+'/articles')
+      .then((response) => {
+        // This current code will dispatch the FETCH_ARTICLE_REJECtED because the API still has null values
+        if (articleIsValid(response.data)) {
+          dispatch({
+            type: t.FETCH_ARTICLE_FULFILLED,
+            payload: response.data,
+          })
+        } else {
+          dispatch({
+            type: t.FETCH_ARTICLE_REJECTED,
+            payload: 'Invalid article array'
+          })
+        }
+      })
+      .catch((err) => {
+        dispatch({
+          type: t.FETCH_ARTICLE_REJECTED,
+          payload: err
+        })
+      })
+  };
 };
+const keyIsValid = (articleObject, key, type) => (
+  key in articleObject) && (typeof (articleObjecty[ key ]) === type

```

Comment:
articleObjecty. Also while this *technically* runs, I'd add another set of parenthesis. Like
const keyIsValikd = () => (
(key in ArticleObject) && (typeof ....)
)